core/src/main/java/hudson/model/Job.java

```diff
@@ -1509,4 +1511,18 @@ public BuildTimelineWidget getTimeline() {
     }
 
     private final static HexStringConfidentialKey SERVER_COOKIE = new HexStringConfidentialKey(Job.class,"serverCookie",16);
+    
+    /**
+     * This handles the situation when Parameterized build checkbox is checked 
+     * but no parameters are selected. User will be redirected to an error page
+     * with proper error message.
+     * @param jsonProperties
+     * @throws FormException 
+     */
+    private static void checkForEmptyParameters(JSONObject jsonProperties) throws FormException{
+        JSONObject parameterDefinitionProperty = jsonProperties.getJSONObject("hudson-model-ParametersDefinitionProperty");
+        if ((parameterDefinitionProperty.getBoolean("specified") == true)&& !parameterDefinitionProperty.has("parameterDefinitions")) {
+		    throw new FormException(Messages.Hudson_NoParamsSpecified(),"parameterDefinitions");
```

Comment:
As noted in #2589, I think this is overkill. Better to just ignore the checkbox.